### DISCUSSION TITLE
JPA with "Using Native SQL in JPA queries"

### DISCUSSION THREAD

**Discussion by yennor** [querydsl/querydsl#2929] [Q&A]
## Description

In the documentation for JPA it is stated that you can use native queries with jpa in the chapter "Using Native SQL in JPA queries" with a maven example. Without any further explanation I needed like 2 hours to get it working (compiling).

- First, the maven configuration without any further comment wasn't helpful at all. I needed quite a while to realize that queryDSL needs access to a database with the scheme you want the S-Classes for and that that Derby example just doesn't do anything at all. Coming from querydsl-jpa, I guessed the classes would be generated from the Q classes or directly the entities, or maybe the scheme would be created in the derby DB from the entities and then the S classes created... (Using the example with the derby DB, freezes Eclipse, which wasn't very helpful neither ;-) )

- Additionally you need the querydsl-sql dependency.

- Add the end, the classes generated aren't called S<EntityName> as the documentation states, but Q<EntityName>. So they've got the same names as the JPA Q-Classes which is really annoying.

I didn't try out at the end if creating a query actually works, since I've decided I'll just create a view. It's a lot less hazle...

**Comment by jwgmeligmeyling** [comment]
The idea of JPA native queries in Querydsl is that they use the same Q-class static metamodel as the JPA queries do. Although querydsl-sql is necessairy as runtime dependency, its codegen should not be necessairy.

There are already a dozen open issues with regard to documentation, so I am going to close this as duplicate. And hopefully someone ever steps up to actually write te docs as we’re currently entirely community driven. With the majority of existing contributors only interested in SQL and JPQL, the JPA Native Query isn’t something that will gather a lot of interest anytime soon.
